go/libkb/upak_loader.go

```diff
@@ -99,14 +125,42 @@ func (u *CachedUPAKLoader) ClearMemory() {
 // version 5 is still OK for non-reset accounts.
 const UPK2MinorVersionCurrent = keybase1.UPK2MinorVersion_V6
 
-func (u *CachedUPAKLoader) getCachedUPAK(ctx context.Context, uid keybase1.UID, info *CachedUserLoadInfo) (*keybase1.UserPlusKeysV2AllIncarnations, bool) {
+func (u *CachedUPAKLoader) getCachedUPAKFromDB(ctx context.Context, uid keybase1.UID, stubMode StubMode) (ret *keybase1.UserPlusKeysV2AllIncarnations) {
+	var tmp keybase1.UserPlusKeysV2AllIncarnations
+	found, err := u.G().LocalDb.GetInto(&tmp, culDBKeyV2(uid, stubMode))
+
+	// As a nice load-saving hack, we can upgrade V5 minor versions to V6 on load if there are no resets.
+	// We just do this in memory.
+	if found && err == nil && tmp.MinorVersion == keybase1.UPK2MinorVersion_V5 && len(tmp.PastIncarnations) == 0 {
+		u.G().VDL.CLogf(ctx, VLog0, "| upgrade disk cache v%d without resets to v%d", keybase1.UPK2MinorVersion_V5, keybase1.UPK2MinorVersion_V6)
+		tmp.MinorVersion = keybase1.UPK2MinorVersion_V6
+	}
+
+	if err != nil {
+		u.G().Log.CWarningf(ctx, "trouble accessing UserPlusKeysV2AllIncarnations cache: %s", err)
+	} else if !found {
+		u.G().VDL.CLogf(ctx, VLog0, "| missed disk cache")
+	} else if tmp.MinorVersion == UPK2MinorVersionCurrent {
+		u.G().VDL.CLogf(ctx, VLog0, "| hit disk cache (v%d)", tmp.MinorVersion)
+		ret = &tmp
+	} else {
+		u.G().VDL.CLogf(ctx, VLog0, "| found old minor version %d, but wanted %d; will overwrite with fresh UPAK", tmp.MinorVersion, UPK2MinorVersionCurrent)
+	}
+	return ret
+}
```

Comment:
v5 is a year outmoded. This could have a bug or future changes to v6 could leave it with a bug. How about dropping v5->v6 support?

Reply:
great idea. I'll just delete those lines